### ISSUE TITLE
Get tranmission rpc url on site root

### ISSUE THREAD

**Issue by nilankadesilva** [transmission/transmission#899]
Hi,
Is there a way to get my webclient to point to http://transmission.example.com/web instead of http://example.com/transmission/web?

I'm using nginx, and I did try serveral combos of transmssion config and nginx site config :/ Some gives me 409 error, others the "too many redirects" error.

**Comment by lps-rocks** [comment]
I don't think so - You could try setting rpc_url in the config to be empty or just '/' but that may cause strange behavior. (I've never tried it).

Using a reverse proxy doesn't always work correctly - I know transmission-web-control has issues with it too.

**Comment by Xdynix** [comment]
Hi, I have the same situation. After several tries, I found this nginx config works for me. Hope it helps.

```nginx
server {
    listen       80;
    server_name  tr.example.com;

    # If accessing http://tr.example.com/ directly,
    # redirect it to http://tr.example.com/web/.
    # Otherwise the next block will proxy it to http://127.0.0.1:9091/transmission/,
    # And Transmission will returns "301 /transmission/web/", which is unwanted.
    location = / {
        return 301 /web/;  # The slash at the end is important.
    }

    # Proxy "http://tr.example.com/*" to "http://127.0.0.1/transmission/*".
    # In fact the Web Interface only needs "/transmission/web" and "/transmission/rpc".
    location / {
        proxy_pass http://127.0.0.1:9091/transmission/;
        proxy_redirect off;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
        proxy_set_header Host $http_host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $remote_addr;
        proxy_set_header X-Forwarded-Host $remote_addr;
    }
}
```

Now the Web Interface can be accessed from `http://tr.example.com/` or `http://tr.example.com/web/` .

**Comment by teras** [comment]
> Hi, I have the same situation. After several tries, I found this nginx config works for me. Hope it helps.
> 

Just to confirm for future reference, this worked for me too, as noted here also: https://github.com/transmission/transmission/issues/231#issuecomment-1462111580

**Comment by ClaymorePT** [comment]
> Hi, I have the same situation. After several tries, I found this nginx config works for me. Hope it helps.
> 
> ```nginx
> server {
>     listen       80;
>     server_name  tr.example.com;
> 
>     # If accessing http://tr.example.com/ directly,
>     # redirect it to http://tr.example.com/web/.
>     # Otherwise the next block will proxy it to http://127.0.0.1:9091/transmission/,
>     # And Transmission will returns "301 /transmission/web/", which is unwanted.
>     location = / {
>         return 301 /web/;  # The slash at the end is important.
>     }
> 
>     # Proxy "http://tr.example.com/*" to "http://127.0.0.1/transmission/*".
>     # In fact the Web Interface only needs "/transmission/web" and "/transmission/rpc".
>     location / {
>         proxy_pass http://127.0.0.1:9091/transmission/;
>         proxy_redirect off;
>         proxy_http_version 1.1;
>         proxy_set_header Upgrade $http_upgrade;
>         proxy_set_header Connection "upgrade";
>         proxy_set_header Host $http_host;
>         proxy_set_header X-Real-IP $remote_addr;
>         proxy_set_header X-Forwarded-For $remote_addr;
>         proxy_set_header X-Forwarded-Host $remote_addr;
>     }
> }
> ```
> 
> Now the Web Interface can be accessed from `http://tr.example.com/` or `http://tr.example.com/web/` .

This procedure doesn't seem to work for me. Somehow, I see the transmission webpage coming up but after that, nothing.
After checking the developer tools from the browser, I noticed that there are POST requests done to rpc that do not have a response. So it seems that redirects are still not properly done.
